[BUG] mlx-cpu backend has a hard dependncy on a deprecated version of gcc

**Describe the bug**
llm operations in the cpu backend jit-compile with g++; if the invoked g++ is not exactly gcc12, jit compilation fails with
[Compile::eval_cpu] Failed to compile function CV2ISigmoidADV2IMultiplyACEV2OMultiplyDB_VV__6142509188972423790_contiguous: Failed to execute command with return code 1: "g++ -std=c++17 -O3 -Wall -fPIC -shared "/tmp/mlx/0.31.1/cpu/CV2ISigmoidADV2IMultiplyACEV2OMultiplyDB_VV__6142509188972423790_contiguous.cpp" -o "/tmp/mlx/0.31.1/cpu/libCV2ISigmoidADV2IMultiplyACEV2OMultiplyDB_VV__6142509188972423790_contiguous.so" 2>&1", the output is: /tmp/mlx/0.31.1/cpu/CV2ISigmoidADV2IMultiplyACEV2OMultiplyDB_VV__6142509188972423790_contiguous.cpp:6009:20: error: redeclaration of C++ built-in type ‘_Float128’ [-fpermissive]

**To Reproduce**

```bash
mlx_lm.chat --model mlx-community/Qwen3-0.6B-4bit
```

**Expected behavior**
system g++ (currently g++ 14) correctly jit compiles - i don't mind providing a specific g++ but gcc12 is actually deprecated at this point.

**Desktop (please complete the following information):**
 - OS Version: NixOS 25.11 
 - Version 0.31.1

**Additional Context**
swapping gcc with gcc12 via
```bash
nix shell nixpkgs/24.11#gcc12Stdenv.cc
mlx_lm.chat --model mlx-community/Qwen3-0.6B-4bit
```
correctly executes the model and generates tokens.

Do you mind pasting the error line in the file?

`/tmp/mlx/0.31.1/cpu/CV2ISigmoidADV2IMultiplyACEV2OMultiplyDB_VV__6142509188972423790_contiguous.cpp:6009:20: error: redeclaration of C++ built-in type ‘_Float128’ [-fpermissive]`

here's a dump, since im not sure that specific error is helpful:

```bash
❯ g++ --version
g++ (GCC) 15.2.0
Copyright (C) 2025 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

❯ uv run mlx_lm.chat --model mlx-community/Qwen3-0.6B-4bit
Fetching 9 files: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 9/9 [00:00<00:00, 15196.75it/s]
Download complete: : 0.00B [00:00, ?B/s]                                                                                                                                                               | 0/9 [00:00<?, ?it/s]
[INFO] Starting chat session with mlx-community/Qwen3-0.6B-4bit.
The command list:
- 'q' to exit
- 'r' to reset the chat
- 'h' to display these commands
>> hi
Traceback (most recent call last):
  File "~/.venv/bin/mlx_lm.chat", line 10, in <module>
    sys.exit(main())
             ~~~~^^
  File "~/.venv/lib/python3.14/site-packages/mlx_lm/chat.py", line 145, in main
    for response in stream_generate(
                    ~~~~~~~~~~~~~~~^
        model,
        ^^^^^^
    ...<12 lines>...
        prompt_cache=prompt_cache,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
    ):
    ^
  File "~/.venv/lib/python3.14/site-packages/mlx_lm/generate.py", line 716, in stream_generate
    for n, (token, logprobs, from_draft) in enumerate(token_generator):
                                            ~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "~/.venv/lib/python3.14/site-packages/mlx_lm/generate.py", line 706, in <genexpr>
    (token, logprobs, False) for token, logprobs in token_generator
                                                    ^^^^^^^^^^^^^^^
  File "~/.venv/lib/python3.14/site-packages/mlx_lm/generate.py", line 442, in generate_step
    mx.eval([c.state for c in prompt_cache])
    ~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: [Compile::eval_cpu] Failed to compile function CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous: Failed to execute command with return code 1: "g++ -std=c++17 -O3 -Wall -fPIC -shared "/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp" -o "/tmp/mlx/0.31.2/cpu/libCV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.so" 2>&1", the output is: /tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:6009:20: error: redeclaration of C++ built-in type ‘_Float128’ [-fpermissive]
 6009 | typedef __float128 _Float128;
      |                    ^~~~~~~~~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:6010:15: error: redeclaration of C++ built-in type ‘_Float32’ [-fpermissive]
 6010 | typedef float _Float32;
      |               ^~~~~~~~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:6011:16: error: redeclaration of C++ built-in type ‘_Float64’ [-fpermissive]
 6011 | typedef double _Float64;
      |                ^~~~~~~~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:6012:16: error: redeclaration of C++ built-in type ‘_Float32x’ [-fpermissive]
 6012 | typedef double _Float32x;
      |                ^~~~~~~~~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:6013:21: error: redeclaration of C++ built-in type ‘_Float64x’ [-fpermissive]
 6013 | typedef long double _Float64x;
      |                     ^~~~~~~~~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In constructor ‘std::uniform_int_distribution<_IntType>::param_type::param_type(_IntType, _IntType)’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:8039:44: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
 8039 |    do { if (__builtin_is_constant_evaluated() && !bool(_M_a <= _M_b)) __builtin_unreachable(); } while (false);
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘std::_Node_handle_common<_Val, _NodeAlloc>::allocator_type std::_Node_handle_common<_Val, _NodeAlloc>::get_allocator() const’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:17688:42: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
17688 |  do { if (__builtin_is_constant_evaluated() && !bool(!this->empty())) __builtin_unreachable(); } while (false);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In constructor ‘std::_Node_handle_common<_Val, _NodeAlloc>::_Node_handle_common(typename _AllocTraits::pointer, const _NodeAlloc&)’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:17730:42: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
17730 |  do { if (__builtin_is_constant_evaluated() && !bool(__ptr != nullptr)) __builtin_unreachable(); } while (false);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘void std::_Node_handle_common<_Val, _NodeAlloc>::_Optional_alloc::operator=(_NodeAlloc&&)’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:17784:46: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
17784 |      do { if (__builtin_is_constant_evaluated() && !bool(_M_alloc == __alloc)) __builtin_unreachable(); } while (false);
      |               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘void std::_Node_handle_common<_Val, _NodeAlloc>::_Optional_alloc::swap(std::_Node_handle_common<_Val, _NodeAlloc>::_Optional_alloc&)’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:17793:46: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
17793 |      do { if (__builtin_is_constant_evaluated() && !bool(_M_alloc == __other._M_alloc)) __builtin_unreachable(); } while (false);
      |               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘std::_Node_handle<_Key, _Value, _NodeAlloc>::key_type& std::_Node_handle<_Key, _Value, _NodeAlloc>::key() const’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:17825:42: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
17825 |  do { if (__builtin_is_constant_evaluated() && !bool(!this->empty())) __builtin_unreachable(); } while (false);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘std::_Node_handle<_Key, _Value, _NodeAlloc>::mapped_type& std::_Node_handle<_Key, _Value, _NodeAlloc>::mapped() const’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:17831:42: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
17831 |  do { if (__builtin_is_constant_evaluated() && !bool(!this->empty())) __builtin_unreachable(); } while (false);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘std::_Node_handle<_Value, _Value, _NodeAlloc>::value_type& std::_Node_handle<_Value, _Value, _NodeAlloc>::value() const’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:17899:42: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
17899 |  do { if (__builtin_is_constant_evaluated() && !bool(!this->empty())) __builtin_unreachable(); } while (false);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘std::_Hashtable<_Key, _Value, _Alloc, _ExtractKey, _Equal, _Hash, _RangeHash, _Unused, _RehashPolicy, _Traits>::insert_return_type std::_Hashtable<_Key, _Value, _Alloc, _ExtractKey, _Equal, _Hash, _RangeHash, _Unused, _RehashPolicy, _Traits>::_M_reinsert_node(node_type&&)’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:18489:46: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
18489 |      do { if (__builtin_is_constant_evaluated() && !bool(get_allocator() == __nh.get_allocator())) __builtin_unreachable(); } while (false);
      |               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘std::_Hashtable<_Key, _Value, _Alloc, _ExtractKey, _Equal, _Hash, _RangeHash, _Unused, _RehashPolicy, _Traits>::iterator std::_Hashtable<_Key, _Value, _Alloc, _ExtractKey, _Equal, _Hash, _RangeHash, _Unused, _RehashPolicy, _Traits>::_M_reinsert_node_multi(const_iterator, node_type&&)’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:18514:42: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
18514 |  do { if (__builtin_is_constant_evaluated() && !bool(get_allocator() == __nh.get_allocator())) __builtin_unreachable(); } while (false);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘void std::_Hashtable<_Key, _Value, _Alloc, _ExtractKey, _Equal, _Hash, _RangeHash, _Unused, _RehashPolicy, _Traits>::_M_merge_unique(_Compatible_Hashtable&)’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:18565:44: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
18565 |    do { if (__builtin_is_constant_evaluated() && !bool(get_allocator() == __src.get_allocator())) __builtin_unreachable(); } while (false);
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘void std::_Hashtable<_Key, _Value, _Alloc, _ExtractKey, _Equal, _Hash, _RangeHash, _Unused, _RehashPolicy, _Traits>::_M_merge_multi(_Compatible_Hashtable&)’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:18590:44: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
18590 |    do { if (__builtin_is_constant_evaluated() && !bool(get_allocator() == __src.get_allocator())) __builtin_unreachable(); } while (false);
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::node_type std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::extract(const_iterator)’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:19860:42: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
19860 |  do { if (__builtin_is_constant_evaluated() && !bool(__pos != end())) __builtin_unreachable(); } while (false);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘std::unordered_multimap<_Key, _Tp, _Hash, _Pred, _Alloc>::node_type std::unordered_multimap<_Key, _Tp, _Hash, _Pred, _Alloc>::extract(const_iterator)’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:20336:42: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
20336 |  do { if (__builtin_is_constant_evaluated() && !bool(__pos != end())) __builtin_unreachable(); } while (false);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘void std::vector<_Tp, _Alloc>::swap(std::vector<_Tp, _Alloc>&)’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:21893:42: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
21893 |  do { if (__builtin_is_constant_evaluated() && !bool(_Alloc_traits::propagate_on_container_swap::value || _M_get_Tp_allocator() == __x._M_get_Tp_allocator())) __builtin_unreachable(); } while (false);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘void std::vector<bool, _Alloc>::swap(std::vector<bool, _Alloc>&)’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:22812:42: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
22812 |  do { if (__builtin_is_constant_evaluated() && !bool(_Bit_alloc_traits::propagate_on_container_swap::value || _M_get_Bit_allocator() == __x._M_get_Bit_allocator())) __builtin_unreachable(); } while (false);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::const_reference std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::operator[](size_type) const’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:34950:42: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
34950 |  do { if (__builtin_is_constant_evaluated() && !bool(__pos <= size())) __builtin_unreachable(); } while (false);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::reference std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::operator[](size_type)’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:34956:42: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
34956 |  do { if (__builtin_is_constant_evaluated() && !bool(__pos <= size())) __builtin_unreachable(); } while (false);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::reference std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::front()’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:34979:42: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
34979 |  do { if (__builtin_is_constant_evaluated() && !bool(!empty())) __builtin_unreachable(); } while (false);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::const_reference std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::front() const’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:34985:42: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
34985 |  do { if (__builtin_is_constant_evaluated() && !bool(!empty())) __builtin_unreachable(); } while (false);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::reference std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::back()’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:34991:42: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
34991 |  do { if (__builtin_is_constant_evaluated() && !bool(!empty())) __builtin_unreachable(); } while (false);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::const_reference std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::back() const’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:34997:42: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
34997 |  do { if (__builtin_is_constant_evaluated() && !bool(!empty())) __builtin_unreachable(); } while (false);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In member function ‘void std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::pop_back()’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:35263:42: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
35263 |  do { if (__builtin_is_constant_evaluated() && !bool(!empty())) __builtin_unreachable(); } while (false);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In function ‘typename __gnu_cxx::__enable_if<std::__is_char<_CharT>::__value, void>::__type std::advance(istreambuf_iterator<_CharT, char_traits<_CharT> >&, _Distance)’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:39746:47: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
39746 |       do { if (__builtin_is_constant_evaluated() && !bool(__n > 0)) __builtin_unreachable(); } while (false);
      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In function ‘std::complex<_Tp> std::polar(const _Tp&, const _Tp&)’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:44799:47: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
44799 |       do { if (__builtin_is_constant_evaluated() && !bool(__rho >= 0)) __builtin_unreachable(); } while (false);
      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In instantiation of ‘std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::reference std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::operator[](size_type) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; reference = char&; size_type = long unsigned int]’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:36370:46:   required from here
36370 |     __detail::__to_chars_10_impl(&__str[__neg], __len, __uval);
      |                                              ^
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:34956:42: warning: ‘std::is_constant_evaluated’ always evaluates to false in a non-‘constexpr’ function [-Wtautological-compare]
34956 |  do { if (__builtin_is_constant_evaluated() && !bool(__pos <= size())) __builtin_unreachable(); } while (false);
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In instantiation of ‘mlx::core::simd::Simd<float, N> mlx::core::detail::FromFP8::operator()(mlx::core::simd::Simd<unsigned char, N>) [with int N = 1]’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:46232:19:   required from here
46232 |     return (*this)(Simd<uint8_t, 1>(x)).value;
      |            ~~~~~~~^~~~~~~~~~~~~~~~~~~~~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:46220:25: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
46220 |       auto converted = *(Simd<float16_t, N>*)(&v);
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:46220:25: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp: In instantiation of ‘mlx::core::simd::Simd<T, N> mlx::core::simd::exp(Simd<T, N>) [with T = mlx::core::_MLX_BFloat16; int N = 1]’:
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:46150:38:   required from ‘mlx::core::simd::Simd<T, N> mlx::core::detail::Sigmoid::operator()(mlx::core::simd::Simd<T, N>) [with int N = 1; T = mlx::core::_MLX_BFloat16]’
46150 |     auto y = 1.0f / (1.0f + simd::exp(simd::abs(x)));
      |                             ~~~~~~~~~^~~~~~~~~~~~~~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:46153:59:   required from ‘T mlx::core::detail::Sigmoid::operator()(T) [with T = mlx::core::_MLX_BFloat16]’
46153 |   template <typename T> T operator()(T x) { return (*this)(Simd<T, 1>(x)).value; }
      |                                                    ~~~~~~~^~~~~~~~~~~~~~~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:46307:31:   required from here
46307 |   bfloat16_t tmp_D = Sigmoid()(tmp_A);
      |                      ~~~~~~~~~^~~~~~~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:45989:51: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
45989 |     auto result = select(isnan(x_init), x_init, (*(Simd<float, N>*)&epart) * x);
      |                                                   ^~~~~~~~~~~~~~~~~~~~~~~
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:45989:51: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
/tmp/mlx/0.31.2/cpu/CV2ISigmoidADV2IBroadcastACEV2IBroadcastCAFV2IMultiplyDEGV2IBroadcastFBHV2IBroadcastBFIV2OMultiplyGH_VV_V2V2_6142509188972423790_contiguous.cpp:45870:65: note: initializing argument 1 of ‘mlx::core::simd::Simd<decltype ((a.value * b.value)), 1> mlx::core::simd::operator*(Simd<T, 1>, Simd<T2, 1>) [with T1 = float; T2 = float; decltype ((a.value * b.value)) = float]’
45870 | template <typename T1, typename T2> auto operator *(Simd<T1, 1> a, Simd<T2, 1> b) ->Simd<decltype(a.value * b.value), 1> { return a.value * b.value; } template <typename T1, typename T2> auto operator *(T1 a, Simd<T2, 1> b)->Simd<decltype(a * b.value), 1> { return a * b.value; } template <typename T1, typename T2> auto operator *(Simd<T1, 1> a, T2 b)->Simd<decltype(a.value * b), 1> { return a.value * b; }
      |                                                     ~~~~~~~~~~~~^
```

--- a/mlx/backend/cpu/compiled.cpp
+++ b/mlx/backend/cpu/compiled.cpp
@@ -10,7 +10,6 @@
 #include <fmt/format.h>
 
 #include "mlx/backend/common/compiled.h"
-#include "mlx/backend/cpu/compiled_preamble.h"
 #include "mlx/backend/cpu/encoder.h"
 #include "mlx/backend/cpu/jit_compiler.h"
 #include "mlx/device.h"
@@ -316,7 +315,9 @@ void Compiled::eval_cpu(
   // Get the function
   auto fn_ptr = compile(kernel_name, [&, contiguous = contiguous]() {
     std::ostringstream kernel;
-    kernel << get_kernel_preamble() << std::endl;
+    kernel << std::get<2>(JitCompiler::get_preamble()) << std::endl;
+    kernel << "using namespace mlx::core;" << std::endl;
+    kernel << "using namespace mlx::core::detail;" << std::endl;
     kernel << "extern \"C\"  {" << std::endl;
     build_kernel(
         kernel,

--- a/mlx/backend/cpu/compiled_preamble.h
+++ b/mlx/backend/cpu/compiled_preamble.h
@@ -9,4 +9,4 @@
 #include "mlx/backend/cpu/binary_ops.h"
 // clang-format on
 
-const char* get_kernel_preamble();
+const char* get_prebuilt_preamble();

--- a/mlx/backend/cpu/jit_compiler.cpp
+++ b/mlx/backend/cpu/jit_compiler.cpp
@@ -1,6 +1,8 @@
 // Copyright © 2024 Apple Inc.
 
 #include "mlx/backend/cpu/jit_compiler.h"
+#include "mlx/backend/common/utils.h"
+#include "mlx/backend/cpu/compiled_preamble.h"
 
 #include <algorithm>
 #include <sstream>
@@ -86,30 +88,61 @@ const VisualStudioInfo& GetVisualStudioInfo() {
 
 #endif // _MSC_VER
 
+const std::tuple<bool, std::string, std::string>& JitCompiler::get_preamble() {
+  static auto preamble = []() -> std::tuple<bool, std::string, std::string> {
+    // Check whether the headers are shipped with the binary, if so use the
+    // preamble from the headers, otherwise use the prebuilt one embeded in
+    // binary, which may not work with all compilers.
+    auto root_dir = current_binary_dir();
+#if !defined(_WIN32)
+    root_dir = root_dir.parent_path();
+#endif
+    auto include_dir = root_dir / "include";
+    if (std::filesystem::exists(include_dir / "mlx")) {
+      return std::make_tuple(
+          true,
+          include_dir.string(),
+          "#include \"mlx/backend/cpu/compiled_preamble.h\"\n");
+    } else {
+      return std::make_tuple(false, "", get_prebuilt_preamble());
+    }
+  }();
+  return preamble;
+}
+
 std::string JitCompiler::build_command(
     const std::filesystem::path& dir,
     const std::string& source_file_name,
     const std::string& shared_lib_name) {
+  auto& [use_include, include_dir, preamble] = get_preamble();
 #ifdef _MSC_VER
+  std::string extra_flags;
+  if (use_include) {
+    extra_flags += fmt::format("/I \"{}\"", include_dir);
+  }
   const VisualStudioInfo& info = GetVisualStudioInfo();
-  std::string libpaths;
   for (const std::string& lib : info.libpaths) {
-    libpaths += fmt::format(" /libpath:\"{0}\"", lib);
+    extra_flags += fmt::format(" /libpath:\"{}\"", lib);
   }
   return fmt::format(
       "\""
-      "cd /D \"{0}\" && "
-      "\"{1}\" /LD /EHsc /MD /Ox /nologo /std:c++17 \"{2}\" "
-      "/link /out:\"{3}\" {4} 2>&1"
+      "cd /D \"{}\" && "
+      "\"{}\" /LD /EHsc /MD /Ox /nologo /std:c++17 {} \"{}\" "
+      "/link /out:\"{}\" 2>&1"
       "\"",
       dir.string(),
       info.cl_exe,
+      extra_flags,
       source_file_name,
-      shared_lib_name,
-      libpaths);
+      shared_lib_name);
 #else
+  std::string extra_flags;
+  if (use_include) {
+    extra_flags = fmt::format("-I \"{}\"", include_dir);
+  }
   return fmt::format(
-      "g++ -std=c++17 -O3 -Wall -fPIC -shared \"{0}\" -o \"{1}\" 2>&1",
+      "g++ -std=c++17 -O3 -Wall -fPIC -shared {} \"{}\" -o \"{}\" 2>&1",
+      extra_flags,
       (dir / source_file_name).string(),
       (dir / shared_lib_name).string());
 #endif

--- a/mlx/backend/cpu/jit_compiler.h
+++ b/mlx/backend/cpu/jit_compiler.h
@@ -7,6 +7,9 @@ namespace mlx::core {
 
 class JitCompiler {
  public:
+  // Return the includes that should be prepended to the source code.
+  static const std::tuple<bool, std::string, std::string>& get_preamble();
+
   // Build a shell command that compiles a source code file to a shared library.
   static std::string build_command(
       const std::filesystem::path& dir,

--- a/mlx/backend/cpu/make_compiled_preamble.ps1
+++ b/mlx/backend/cpu/make_compiled_preamble.ps1
@@ -15,13 +15,6 @@ $CONTENT = $CONTENT | Where-Object { $_.Trim() -ne '' }
 # Concatenate to string.
 $CONTENT = $CONTENT -join "`n"
 
-# Append extra content.
-$CONTENT = @"
-$($CONTENT)
-using namespace mlx::core;
-using namespace mlx::core::detail;
-"@
-
 # Convert each char to ASCII code.
 # Unlike the unix script that outputs string literal directly, the output from
 # MSVC is way too large to be embedded as string and compilation will fail, so
@@ -29,7 +22,7 @@ using namespace mlx::core::detail;
 $CHARCODES = ([System.Text.Encoding]::ASCII.GetBytes($CONTENT) -join ', ') + ', 0'
 
 $OUTPUT = @"
-const char* get_kernel_preamble() {
+const char* get_prebuilt_preamble() {
   static char preamble[] = { $CHARCODES };
   return preamble;
 }

--- a/mlx/backend/cpu/make_compiled_preamble.sh
+++ b/mlx/backend/cpu/make_compiled_preamble.sh
@@ -30,12 +30,10 @@ fi
 CONTENT=$($GCC $CC_FLAGS -I "$SRCDIR" -E -P "$SRCDIR/mlx/backend/cpu/compiled_preamble.h" 2>/dev/null)
 
 cat << EOF > "$OUTPUT_FILE"
-const char* get_kernel_preamble() {
+const char* get_prebuilt_preamble() {
 return R"preamble(
 $INCLUDES
 $CONTENT
-using namespace mlx::core;
-using namespace mlx::core::detail;
 )preamble";
 }
 EOF

